### ISSUE TITLE
unknown GLU entry from ghci

### ISSUE THREAD

**Issue by rgoggans** [haskell-opengl/GLURaw#8]
Trying to run some library example code. When I do stack build and run the executable everything works perfectly with a working GUI. 

Unfortunately when I try to call my GUI function from stack ghci, it crashes with this: 

`user error (unknown GLU entry gluOrtho2D)`

Using
OpenGL-3.0.1.0
GLUT-2.7.0.10
GLURaw-2.0.0.2
OpenGLRaw-3.2.1.0

on Arch Linux

any help is appreciated

**Comment by svenpanne** [comment]
I can reproduce this both with LTS Haskell (based on GHC 7.10.3) and Stackage nightly (based on GHC 8.0.1) on Ubuntu 16.04, too. This is somehow related to GHCi's dynamic linker, which has changed a few times in the past, and there are still issues with it. Alas, I don't remember the details, but perhaps you can get some answers on the `haskell-cafe` mailing list. All I can say is that things worked even within GHCi at some point in the past... :wink:

The gory details: Given an API entry name like `"gluOrtho2D"`, the GLURaw package uses `dlopen` and `dlsym` to get the address of the function, see the [Haskell part](https://github.com/haskell-opengl/GLURaw/blob/2fbff384e9637e1c1763c65a5f843b8fc36b1fea/src/Graphics/GLU/Functions.hs#L91) and the [C part](https://github.com/haskell-opengl/GLURaw/blob/2fbff384e9637e1c1763c65a5f843b8fc36b1fea/cbits/HsGLURaw.c#L77). For some reason, `dlsym` doesn't find the GLU entries when running under GHCi, while the compiled version does.

If you manage to find out the reason and/or a workaround, I would be glad to hear it.

**Comment by rgoggans** [comment]
thank you for the quick and thorough reply! I sent out an email to haskell-cafe

For now I picked a random old version of GHC and things are working! I'm using 7.8.4 via LTS-1.0

**Comment by nh2** [comment]
I get this too, in the shape of 

```
<interactive>: user error (unknown GLU entry gluPerspective)
<interactive>: interrupted
```

**Comment by svenpanne** [comment]
I think I've fixed this problem via 54e3c2a9fb0acc06094f039f1207feaf7f942d77, a new version has already been uploaded to Hackage, so it should be in Stackackge nightly soon, too.

If not: Just re-open this issue...
